vector-im/element-android
# [Rich text editor] Add list formatting buttons to the rich text editor

## Type of change


## Content

Add list formatting buttons to the rich text editor.

## Motivation and context

[`PSU-1058`](https://element-io.atlassian.net/browse/PSU-1058)

## Screenshots / GIFs

[lists.webm](https://user-images.githubusercontent.com/4940864/210537572-bc411733-19ce-4f74-8ebd-a54850c9fc77.webm)

### Toggling list types
[switching-list-types.webm](https://user-images.githubusercontent.com/4940864/210573528-023943ae-aeaf-44ec-95a7-1c138eedc719.webm)


## Tests

Note that the feature is still a little unstable so expect some quirks while testing manually.

- Enable rich text editor in labs
- Start composing a message
- Tap the 'bullet list' button
- _Note the text becomes a bullet list_
- Tap the 'numbered list' button
- _Note the text becomes a numbered list_

## Tested devices

- OS version(s): Android 13

## Checklist



[PSU-1058]: https://element-io.atlassian.net/browse/PSU-1058?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PSU-1058]: https://element-io.atlassian.net/browse/PSU-1058?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
